app/scripts/modules/core/src/search/widgets/SearchFilterTypeRegistry.ts

```diff
@@ -0,0 +1,47 @@
+export interface IFilterType {
+  key: string;
+  modifier: string;
+  text: string;
+}
+
+export class SearchFilterTypeRegistry {
+
+  public static KEYWORD_FILTER: IFilterType = Object.freeze({
+    key: 'keyword',
+    modifier: 'key',
+    text: 'Keyword'
+  });
+
+  public static NAME_FILTER: IFilterType = Object.freeze({
+    key: 'name',
+    modifier: 'name',
+    text: 'Name'
+  });
+
+  private static registry: Map<string, IFilterType> = new Map<string, IFilterType>();
+
+  public static register(filterType: IFilterType): void {
+    const { key, modifier, text } = filterType;
+    SearchFilterTypeRegistry.registry.set(filterType.key, {
+      key, text, modifier: modifier.toLocaleLowerCase()
+    });
+  }
+
+  public static getFilterType(key: string): IFilterType {
+
+    let result = SearchFilterTypeRegistry.registry.get(key);
+    if (key === SearchFilterTypeRegistry.NAME_FILTER.key) {
+      result = SearchFilterTypeRegistry.NAME_FILTER;
+    }
```

Comment:
Is there a reason to not put keyword and name into the registry? Also, any reason to exclude keyword checking here?